frontend/js/src/settings/missing-data/MissingMBData.tsx

```diff
@@ -266,137 +287,120 @@ export default class MissingMBDataPage extends React.Component<
+                    tabIndex={0}
+                  >
+                    &larr; Previous
+                  </a>
+                </li>
+                <li
+                  className={`next ${
+                    currPage && currPage >= totalPages ? "hidden" : ""
+                  }`}
+                  style={{ marginLeft: "auto" }}
                 >
-                  Next &rarr;
-                </a>
-              </li>
-            </ul>
+                  <a
+                    role="button"
+                    onClick={this.handleClickNext}
+                    onKeyDown={(e) => {
+                      if (e.key === "Enter") this.handleClickNext();
+                    }}
+                    tabIndex={0}
+                  >
+                    Next &rarr;
+                  </a>
+                </li>
+              </ul>
+            </div>
           </div>
+          <BrainzPlayer
+            listens={missingMBDataAsListen}
+            listenBrainzAPIBaseURI={APIService.APIBaseURI}
+            refreshSpotifyToken={APIService.refreshSpotifyToken}
+            refreshYoutubeToken={APIService.refreshYoutubeToken}
+            refreshSoundcloudToken={APIService.refreshSoundcloudToken}
+          />
         </div>
-        <BrainzPlayer
-          listens={missingMBDataAsListen}
-          listenBrainzAPIBaseURI={APIService.APIBaseURI}
-          refreshSpotifyToken={APIService.refreshSpotifyToken}
-          refreshYoutubeToken={APIService.refreshYoutubeToken}
-          refreshSoundcloudToken={APIService.refreshSoundcloudToken}
-        />
-      </div>
+      </>
     );
   }
 }
 
-document.addEventListener("DOMContentLoaded", () => {
-  const {
-    domContainer,
-    reactProps,
-    globalAppContext,
-    sentryProps,
-  } = getPageProps();
-  const { sentry_dsn, sentry_traces_sample_rate } = sentryProps;
-
-  if (sentry_dsn) {
-    Sentry.init({
-      dsn: sentry_dsn,
-      integrations: [new Integrations.BrowserTracing()],
-      tracesSampleRate: sentry_traces_sample_rate,
-    });
-  }
-
-  const { missingData, user } = reactProps;
+export function MissingMBDataPageWrapper() {
+  const data = useLoaderData() as MissingMBDataProps;
+  const { currentUser: user } = React.useContext(GlobalAppContext);
+  return <MissingMBDataPage {...data} user={user} />;
+}
 
-  const MissingMBDataPageWithAlertNotification = withAlertNotifications(
-    MissingMBDataPage
-  );
-  const renderRoot = createRoot(domContainer!);
-  renderRoot.render(
-    <ErrorBoundary>
-      <GlobalAppContext.Provider value={globalAppContext}>
-        <NiceModal.Provider>
-          <MissingMBDataPageWithAlertNotification
-            missingData={missingData}
-            user={user}
-          />
-        </NiceModal.Provider>
-      </GlobalAppContext.Provider>
-    </ErrorBoundary>
-  );
-});
+export const MissingMBDataPageLoader = async () => {
+  const response = await fetch("/profile/missing-data/", {
+    method: "POST",
```

Comment:
I don't understand why these requests to fetch data are POST rather than GET requests.
Could you walk me through it?

Reply:
Since the routing is now being handled by react-router-dom, the GET request to the URL now return the page, and the POST requests now return the initial props which are essential for the page.